nemo_skills/pipeline/generate.py

```diff
@@ -557,126 +194,95 @@ def generate(
         chunk_ids = [None]
 
     # Prepare cluster config and mount paths
-    cluster_config = get_cluster_config(cluster, config_dir)
-    cluster_config = resolve_mount_paths(cluster_config, mount_paths, create_remote_dir=check_mounted_paths)
+    cluster_config = pipeline_utils.get_cluster_config(cluster, config_dir)
+    cluster_config = pipeline_utils.resolve_mount_paths(
+        cluster_config, mount_paths, create_remote_dir=check_mounted_paths
+    )
 
     if not log_dir:
         log_dir = f"{output_dir}/generation-logs"
 
-    output_dir, log_dir = check_mounts(
+    output_dir, log_dir = pipeline_utils.check_mounts(
         cluster_config,
         log_dir=log_dir,
         mount_map={output_dir: None},
         check_mounted_paths=check_mounted_paths,
     )
 
-    get_server_command = server_command_factories[generation_type]
-    get_cmd = client_command_factories[generation_type]
-    cmd_script = client_command_scripts[generation_type]
     original_server_address = server_address
 
-    # If GenerationType is `generate`, check if custom GenerationTask is provided via ctx.obj['generation_task_type']
-    if (
-        generation_type == GenerationType.generate
-        and ctx.obj is not None
-        and isinstance(ctx.obj, dict)
-        and 'generation_task_type' in ctx.obj
-    ):
-        generation_task = ctx.obj['generation_task_type']  # type: type(GenerationTask)
-        assert issubclass(
-            generation_task, GenerationTask
-        ), f"`generation_task_type` must be a subclass of GenerationTask"
-        cmd_script = generation_task.get_generation_module()
-        cmd_extra_args = generation_task.get_generation_default_args()
-        cmd_script = f"{cmd_script.strip()} {cmd_extra_args.strip()}"
+    if generation_module is not None and generation_type is not None:
+        raise ValueError("Cannot specify both generation_module and generation_type. ")
+    if generation_module is None:
+        generation_module = GENERATION_MODULE_MAP[generation_type or GenerationType.generate]
 
+    generation_task = importlib.import_module(generation_module)
```

Comment:
Need a `try/except` block to catch ill-specified `generation_module`

Reply:
why? It will just fail with a clear error in that case